Todo/AppDelegate.swift

```diff
@@ -0,0 +1,420 @@
+//
+//  AppDelegate.swift
+//  Todo
+//
+//  Created by Pasin Suriyentrakorn on 2/8/16.
+//  Copyright © 2016 Couchbase. All rights reserved.
+//
+
+import UIKit
+
+let kLoginFlowEnabled = true
+let kEncryptionEnabled = true
+let kSyncEnabled = true
+let kSyncGatewayUrl = NSURL(string: "http://10.17.2.133:4984/todo/")!
+let kLoggingEnabled = true
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate, LoginViewControllerDelegate {
+    var window: UIWindow?
+
+    var database: CBLDatabase!
+    var pusher: CBLReplication!
+    var puller: CBLReplication!
+    var syncError: NSError?
+    var conflictsLiveQuery: CBLLiveQuery?
+
+    // MARK: - Application Life Cycle
+    
+    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
+        if kLoggingEnabled {
+            enableLogging()
+        }
+        
+        if kLoginFlowEnabled {
+            login()
+        } else {
+            do {
+                try startSession(withUsername: "todo")
+            } catch let error as NSError {
+                NSLog("Cannot start a session: %@", error)
+                return false
+            }
+        }
+        
+        return true
+    }
+    
+    // MARK: - Logging
+    func enableLogging() {
+        CBLManager.enableLogging("CBLDatabase")
+        CBLManager.enableLogging("View")
+        CBLManager.enableLogging("ViewVerbose")
+        CBLManager.enableLogging("Query")
+        CBLManager.enableLogging("Sync")
+        CBLManager.enableLogging("SyncVerbose")
+    }
+    
+    // MARK: - Session
+    
+    func startSession(withUsername username:String, withPassword password:String? = nil,
+        withNewPassword newPassword:String? = nil) throws {
+            try openDatabase(withUsername: username, withKey: password, withNewKey: newPassword)
+            Session.username = username
+            startReplication(withUsername: username, andPassword: newPassword ?? password)
+            showApp()
+    }
+    
+    func openDatabase(withUsername username:String, withKey key:String?,
+        withNewKey newKey:String?) throws {
+            let dbname = username
+            let options = CBLDatabaseOptions()
+            options.create = true
+
+            if kEncryptionEnabled {
+                if let encryptionKey = key {
```

Comment:
I don't think this `if let` is necessary — `options.encryptionKey` is already an optional, so you should be able to assign `key` directly to it.

…Oh, you can't because I forgot to mark that property as `nullable`. Crap. I just filed a bug to remind myself to fix that.